arrows/burnout/burnout_image_enhancer.cxx

```diff
@@ -0,0 +1,242 @@
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither name of Kitware, Inc. nor the names of any contributors may be used
+ *    to endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "burnout_image_enhancer.h"
+
+#include <string>
+#include <sstream>
+#include <exception>
+
+#include <arrows/vxl/image_container.h>
+
+#include <vital/exceptions.h>
+
+#include <video_transforms/video_enhancement_process.h>
+
+
+namespace kwiver {
+namespace arrows {
+namespace burnout {
+
+typedef vidtk::video_enhancement_process< vxl_byte > process_8bit;
+typedef vidtk::video_enhancement_process< vxl_uint_16 > process_16bit;
+
+
+// ==================================================================================
+class burnout_image_enhancer::priv
+{
+public:
+  priv()
+    : m_config_file( "burnout_enhancer.conf" )
+    , is_16bit_mode( true )
+  {}
+
+  ~priv()
+  {}
+
+  // Items from the config
+  std::string m_config_file;
+
+  // VIDTK config block with parameters
+  vidtk::config_block m_vidtk_config;
+  bool is_16bit_mode;
+
+  // Pointer to VIDTK process
+  std::unique_ptr< vidtk::process > m_process;
+
+  // Configure internal process
+  void configure_process( bool for_16bit = false );
+
+  vital::logger_handle_t m_logger;
+};
+
+
+// ==================================================================================
+burnout_image_enhancer
+::burnout_image_enhancer()
+  : d( new priv() )
+{
+}
+
+
+burnout_image_enhancer
+::~burnout_image_enhancer()
+{
+}
+
+// ----------------------------------------------------------------------------------
+
+void
+burnout_image_enhancer::priv
+::configure_process( bool for_16bit )
+{
+  if( for_16bit == is_16bit_mode )
+  {
+    return;
+  }
+
+  if( for_16bit )
+  {
+    process_16bit* process = new process_16bit( "filter" );
+    m_process.reset( process );
+    is_16bit_mode = true;
+
+    if( !process->set_params( m_vidtk_config ) )
+    {
+      throw std::runtime_error(  "Failed to set pipeline parameters" );
+    }
+
+    if( !process->initialize() )
+    {
+      throw std::runtime_error( "Failed to initialize pipeline" );
```

Comment:
Shouldn't all of these throws be VITAL_THROW?

Reply:
VITAL_THROW only works for exceptions derived from vital_exception :-(